Blog for the chinese localization (#2237)
jenkins-infra/jenkins.io: content/blog/2019/05/2019-05-09-chinese-localization.adoc

is the open source community way. We got a lot of ideas for the localization from there. As a result,
we achieved some goals that I'd like to share here.

== JEP-216

Previously all language files were aggregated in one place for core and for each plugin.
For https://github.com/jenkinsci/jep/blob/master/jep/216/README.adoc[this proposal], each language has a single localization plugin, such as https://github.com/jenkinsci/localization-zh-cn-plugin[Chinese Localization plugin].
Finally, https://github.com/jenkinsci/localization-support-plugin[Localization Support Plugin] and
https://github.com/jenkinsci/localization-zh-cn-plugin[Chinese Localization plugin] are able to support
all kinds of existed i10n resource files. We removed all Chinese localization files at the https://github.com/jenkinsci/jenkins/pull/4008[PR-4008].

I really appreciate https://github.com/daniel-beck[Daniel Beck] for helping me to add localization support,
https://github.com/bitwiseman[Liam Newman] helping me to review JEP-216, and many other community members.

== Chinese Localization SIG

We believe that this SIG could help to improve Jenkins experience for Chinese users and gather more contributors

> who're from China. This SIG will responsible for maintaining the https://jenkins.io/zh/[Chinese Jenkins website],

Reviewer: will --> are ?

---
:layout: post
:title: A Big Step of the Chinese Localization
:tags:
- general
- community
- localization
- chinese
:author: linuxsuren
---

Since 2017, I started to do some contributions to the Jenkins community. As a beginner, translation might be

> the easiest way to involve. You don't need to understand the whole context, even to create a ticket in the

Reviewer 2: ```suggestion
the easiest way to help the project. You don't need to understand the whole context, even to create a ticket in the
```

issue tracker system. Improvement of localization usually is minor. But some problems occurred soon,

> there isn't a native speaker of Chinese could help to do the real review for my PRs. So, sometimes my PRs

Reviewer: 这两行是不是用过去式比较好？

Reviewer: there wasn't a native speaker
were hard to be merged into master

Author: fixed with another way

> there isn't a native speaker of Chinese could help to do the real review for my PRs. So, sometimes my PRs

Reviewer 2: ```suggestion
there isn't a native speaker of Chinese that could review my PRs. So, sometimes my PRs
```

> are hard to be merged into master.

Reviewer 2: ```suggestion
are delayed from being merged into master.
```

Some contributors told me that I can start a thread at the mailing list. Normally, discussing at the mailing list
is the open source community way. We got a lot of ideas for the localization from there. As a result,
we achieved some goals that I'd like to share here.

== JEP-216

> Previously all language files were aggregated in one place for core and for each plugin.

Reviewer 2: ```suggestion
Previously, language localization files were distributed in core and in each plugin.
```

For https://github.com/jenkinsci/jep/blob/master/jep/216/README.adoc[this proposal], each language has a single localization plugin, such as https://github.com/jenkinsci/localization-zh-cn-plugin[Chinese Localization plugin].
Finally, https://github.com/jenkinsci/localization-support-plugin[Localization Support Plugin] and
https://github.com/jenkinsci/localization-zh-cn-plugin[Chinese Localization plugin] are able to support

> all kinds of existed i10n resource files. From the https://plugins.jenkins.io/localization-zh-cn[plugins website],

Reviewer 2: ```suggestion
all types of localization resource files. From the https://plugins.jenkins.io/localization-zh-cn[plugins website],
```

> you can see that the installs number already is 13k.

Reviewer: the installs number already is 13k
-->
the number of installation is already 13k
?

Author: Fixed with another way.

> you can see that the installs number already is 13k.

Reviewer 2: ```suggestion
you can see that there are already 13 000 installations.
```

We removed all Chinese localization files at the https://github.com/jenkinsci/jenkins/pull/4008[PR-4008].

I really appreciate https://github.com/daniel-beck[Daniel Beck] for helping me to add localization support,
https://github.com/bitwiseman[Liam Newman] helping me to review JEP-216, and many other community members.

== Chinese Localization SIG

We believe that this SIG could help to improve Jenkins experience for Chinese users and gather more contributors

> who're from China. This SIG is responsible for maintaining the https://jenkins.io/zh/[Chinese Jenkins website],

Reviewer 2: ```suggestion
from China. This SIG is responsible for maintaining the https://jenkins.io/zh/[Chinese Jenkins website],
```

> promoting the Jenkins community in China in the social media with WeChat account. We publish the translated blog

Reviewer 2: ```suggestion
promoting the Jenkins community in China in the social media with WeChat account. We publish translated blog
```

> articles, Jenkins release notes, JAM or other events at the WeChat account. For now, there're 1.8k followers can

Reviewer 2: ```suggestion
articles, Jenkins release notes, JAM or other events at the WeChat account. For now, there are 1800 followers that can
```

:author: linuxsuren
---

Since 2017, I started to do some contributions to the Jenkins community. As a beginner, the translation work might be
the easiest way to involve. You don't need to understand the whole context, even to create a ticket in the

> issues tracker system. Improvement of localization usually is minor. But some problems occurred soon,

Reviewer: issue tracking system ?
